presto-main/src/main/java/com/facebook/presto/sql/planner/PlanBuilder.java

```diff
@@ -92,6 +97,11 @@ public TranslationMap getTranslations()
 
     public PlanBuilder appendProjections(Iterable<Expression> expressions, SymbolAllocator symbolAllocator, PlanNodeIdAllocator idAllocator)
     {
+        return appendProjections(expressions, symbolAllocator, idAllocator, Optional.empty(), Optional.empty(), Optional.empty(), Optional.empty());
```

Comment:
@sopel39 I really don't like this so I'm open to suggestions. The reason why I can't pass into `appendProjections` the grouping expressions already re-written is because in `appendProjections` the code tries to get the type of the expression and of course the newly re-written expressions haven't been analyzed.

I see one better solution than what I have now: pass in a map containing the types of each expression but wanted to see if you had any better ideas.